packages/syft/src/syft/core/node/common/client_manager/dataset_api.py

```diff
@@ -486,8 +523,8 @@ def delete(self, name: str, skip_check: bool = False) -> bool:
                 sys.stdout.write("Asset deletion cancelled.")
                 return False
 
-        DatasetRequestAPI(self.client).delete(
-            dataset_id=self.id, bin_object_id=asset_id
+        DatasetRequestAPI(self.client).del_asset(
+            dataset_id=str(self.id), asset_id=asset_id, skip_checks=True
```

Comment:
Q: Can't we simply pass `self.id` as a string for the `dataset_id`?

Reply:
Just to by pass my py error as self.id type is not defined. But let me find a more concrete solution.

Reply:
Fixed in https://github.com/OpenMined/PySyft/pull/7079/commits/c664423caee20d946a3d9adc0fc55121dbd5fcc7